.pre-commit-config.yaml

```diff
@@ -0,0 +1,8 @@
+# https://pre-commit.com
+repos:
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+      - id: cargo-check
```

Comment:
cargo-check and clippy are executed on everything or just the modified files?

Reply:
I am afraid it will take too long for each commit

Reply:
And it would be great to have "cargo +1.40.0 update" executed too :)

Reply:
> cargo-check and clippy are executed on everything or just the modified files?

Just the files modified by a commit, that is how pre-commit works. Since the entire repository is clean of warnings, it is possible to run the hook on all the files on the Github actions CI pipeline though.

> I am afraid it will take too long for each commit

Why whould it be slow ? It runs almost instantly on my 10 years old desktop.

Reply:
Because I was afraid it would run on all files :)

Reply:
> And it would be great to have "cargo +1.40.0 update" executed too :)

I don't know of any hook that does that. We would most likely have to write one. This is quite unusual though, most Rust codebases track the `Cargo.lock` file, and update it with explicit commits. May I ask what is the rational to exclude this file from the repository ?